src/native/http_request_response.c

```diff
@@ -441,7 +442,7 @@ JNIEXPORT jobject JNICALL Java_software_amazon_awssdk_crt_http_HttpConnection_ht
     struct aws_byte_cursor uri = aws_jni_byte_cursor_from_jstring(env, jni_uri);
     jsize num_headers = (*env)->GetArrayLength(env, jni_headers);
 
-    struct aws_http_header headers[num_headers];
+    AWS_VARIABLE_LENGTH_ARRAY(struct aws_http_header, headers, num_headers);
```

Comment:
Looks good to me, but what was the reason that this wasn't working?

Reply:
MSVC can't handle variable length arrays on the stack.